pkg/apis/extensions/v1alpha1/types_containerruntime.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ Object = (*ContainerRuntime)(nil)
+
+// ContainerRuntimeResource is a constant for the name of the Container Runtime Extension resource.
+const ContainerRuntimeResource = "ContainerRuntime"
+
+// CRINameWorkerLabel is the name of the label describing the CRI name used in this node.
+const CRINameWorkerLabel = "worker.gardener.cloud/cri-name"
+
+// ContainerRuntimeNameWorkerLabel is a label describing a Container Runtime which should be supported on the node.
+const ContainerRuntimeNameWorkerLabel = "containerruntime.worker.gardener.cloud/%s"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ContainerRuntime is a specification for a container runtime resource.
+type ContainerRuntime struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
```

Comment:
Why is `+optional` needed?

Reply:
I think we are doing this for most types, e.g. https://github.com/gardener/gardener/blob/master/pkg/apis/core/v1beta1/types_cloudprofile.go#L30, although it's not needed in general. This was the upstream k/k style of doing it.

Reply:
Do you want me to remove it or keep the current standard?

Reply:
Hm, I don't care too much. For me both is okay.

Reply:
Ok so i keep it like that.